### PR TITLE
Single page   history

### PR DESCRIPTION
**Pull request by ark2002**
- created single page card
- created single page screen
- added single screen to router
- added history services
- added history context
- added history card
- added history screen
- added all new functionality to other pages aswell

### CODE REVIEW

**Review by akash-1318 — COMMENTED** [review]
nicely written code with proper formatting and good use of context and useReducer hooks, good work .

File: src/frontend/components/VideoPlayer/VideoPlayer.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+            <div className="video__options flex--row secondary__font">
+                <h1>{views}</h1>
+                <div className="video__btns flex--row">
+                    {likes.find((video) => video._id === _id) ?
+                        (<span className="material-icons liked" title="unLike" onClick={() => unLikeHandler(_id)}>thumb_up</span>) :
+                        (<span className="material-icons not--liked" title="Like" onClick={() => {
+                            (auth.isAuth) ? likeHandler(video) : navigate("/signin");
+                        }}>thumb_up</span>)}
+                    {watchLater.find((video) => video._id === _id) ?
+                        <button className="btn btn-color--primary btn-font--secondary" onClick={() => navigate("/watchlater")}>go to watch later</button>
```
Comment (sameep-s) [inline src/frontend/components/VideoPlayer/VideoPlayer.js:60]: here it is very difficult to read the code due to some inconsistent indentation.

File: src/frontend/components/VideoPlayer/VideoPlayer.css [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+.video__options {
+    align-items: center;
+    justify-content: space-between;
+}
+
+.video__btns {
+    align-items: center;
+    gap: 2rem;
```
Comment (sameep-s) [inline src/frontend/components/VideoPlayer/VideoPlayer.css:16]: gap is not supported in some old browsers maybe try achieving this with margins and flex

File: src/frontend/components/Navbar/Navbar.js [hunk 3]
```diff
@@ -77,7 +77,7 @@ function Navbar() {
                 <NavLink to="/signin" onClick={() => setListVisibility(!listVisibility)}><li>Sign-In</li></NavLink>
```
Comment (sameep-s) [inline src/frontend/components/Navbar/Navbar.js:77]: indentation could be imporved.

**Review by sameep-s — APPROVED** [review]
Code is good just Added few suggestions.

**Pull request merged**